src/main/java/spring/intro/dao/UserDaoImpl.java

```diff
@@ -45,4 +45,14 @@ public User add(User user) {
             return getAllUsers.getResultList();
         }
     }
+
+    @Override
+    public User get(Long userId) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from User where id = :id", User.class)
+                    .setParameter("id", userId)
+                    .getSingleResult();
```

Comment:
```suggestion
            return session.createQuery("from User where id = :id", User.class)
                    .setParameter("id", userId)
                    .getSingleResult();
```
            return session.get(User.class, id);

Reply:
wow, that's so simple, thanks :)